sklearn/decomposition/tests/test_kernel_pca.py

```diff
@@ -140,6 +140,18 @@ def test_remove_zero_eig():
     assert_equal(Xt.shape, (3, 0))
 
 
+def test_leave_zero_eig():
```

Comment:
Check that no warning is raised (especially zero division warning)

Reply:
Makes sense. I'll push this right away

Reply:
I will push something but I am quite astonished that `assert_no_warnings` can not be used as a context manager: it make the test code quite ugly... Let me know if there is a better way to do it

Reply:
I agree it's not great. If you're interested you can open an issue and ask if this would be a welcome addition (I've no idea, personally I would find it useful)

Reply:
So I pushed something but I am not very happy with it. Indeed in the other PR that I suggest about kPCA, I suggested to raise new warnings when the kernel is badly conditioned. Therefore in this very test, I would like to assert that numpy zero division (or other) warnings are not raised, but I would not like to be too broad because there WILL be a KernelWarning (see #12145)

Reply:
I see, then maybe it's best to use pytest [warning capture](https://docs.pytest.org/en/latest/warnings.html#recwarn) here and just make sure that ZeroDivision wasn't one of the caught warnings